src/components/button/_button.scss

```diff
@@ -281,26 +281,11 @@
   }
 
   .#{$prefix}--btn--danger {
-    @include button-theme--x(transparent, $support-01, $support-01, $support-01, $support-01, $ibm-colors__red-80);
+    @include button-theme--x($support-01, $support-01, $text-04, $hover-danger, $text-04, $active-danger);
 
     &:hover {
-      color: $inverse-01;
       border: $button-border-width solid transparent;
     }
-
-    &:hover > .#{$prefix}--btn__icon {
-      fill: $inverse-01;
```

Comment:
Note that this line + line 287 styles are not required, since we are setting the text color & icon to `$text-04`, which is always white.